### PR TITLE
feat(interview-prep): Converting and transferring Rosetta problems

### PR DESCRIPTION
**Pull request by NitronR**
- [ ] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [ ] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [ ] My pull request targets the `master` branch of freeCodeCamp.
- [ ] None of my changes are plagiarized from another source without proper attribution.
- [ ] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [ ] My changes do not use shortened URLs or affiliate links.

Closes #XXXXX

### CODE REVIEW

**Comment by NitronR** [comment]
@scissorsneedfoodtoo Sorry for the delay, can you check these please?

**Comment by RandellDawson** [comment]
@scissorsneedfoodtoo It is my understanding that Quincy indicated we were not add any new challenges to the current curriculum after we deployed to production back in October.  We were only allowed to fix existing challenge bugs and make improvements to verbiage of existing challenges.  

Anything new was supposed to go through the new curriculum in the other repo.

I remember we wanted to move some challenges around and add a few new ones to fill in some gaps and Quincy said he preferred no further effort was to be spent on new content or rearranging existing challenges.

I will see if I can find the PR/Issue where this was indicated.  I remember we wanted to

**Comment by moT01** [comment]
I might be mistaken - but I thought that the `coding interview prep` section was going to remain around when the projects get released. So maybe it's not a bad idea to consider adding these.

**Comment by scissorsneedfoodtoo** [comment]
Sorry all, I just saw this PR.

@RandellDawson, like @moT01 mentioned, the Coding Interview Prep section will still be around after all the new projects are released. It's fine to continue adding problems to this section, as most will likely be these Rosetta Code ports.

**Comment by NitronR** [comment]
@scissorsneedfoodtoo Hey! Let me know how to proceed with this. Thank you.

**Review by scissorsneedfoodtoo — COMMENTED** [review]
@NitronR, sorry for the delay in reviewing this. Thank you for your patience.

I'll go through the challenges now and leave review comments if I notice anything.

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/last-letter-first-letter.english.md
Comment (scissorsneedfoodtoo) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/last-letter-first-letter.english.md]: I tried to simplify these instructions a bit and remove a bit of redundancy:

Write a function that takes an input array of words. The function should return an array of words where the first letter of each word is the same as the last letter of the previous word. Only use the words in the input array, and once a word is used it cannot be repeated. The words in the return array should be selected and sequenced so that that its length is maximized.

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/last-letter-first-letter.english.md
Comment (scissorsneedfoodtoo) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/last-letter-first-letter.english.md]: This is optional and more of a note for future PRs, but we no longer need the last argument for the `testString`s. This one can be shortened to:

```
testString: assert(Array.isArray(findLongestChain(["certain", "each", "game", "involves", "starting", "with", "word"])));
```

But again, this change is definitely not required for this PR.

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/letter-frequency.english.md
Comment (scissorsneedfoodtoo) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/letter-frequency.english.md]: Maybe this could read:

Given a string, calculate the frequency of each character.

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/levenshtein-distance.md
Comment (scissorsneedfoodtoo) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/levenshtein-distance.md]: We can change the `<b>` tags to strong here. I've also added an `ul` and some `&nbsp;` characters to better mimic the original Rosetta Code description:

```
In information theory and computer science, the <strong>Levenshtein distance</strong> is a <a href="https://en.wikipedia.org/wiki/string metric">metric</a> for measuring the amount of difference between two sequences (i.e. an <a href="https://en.wikipedia.org/wiki/edit distance">edit distance</a>). The Levenshtein distance between two strings is defined as the minimum number of edits needed to transform one string into the other, with the allowable edit operations being insertion, deletion, or substitution of a single character.
Example:
The Levenshtein distance between "<strong>kitten</strong>" and "<strong>sitting</strong>" is 3, since the following three edits change one into the other, and there isn't a way to do it with fewer than three edits:
<ul>
  <li><strong>k</strong>itten &nbsp; <strong>s</strong>itten &nbsp;&nbsp; (substitution of 'k' with 's')</strong></li>
  <li>sitt<strong>e</strong>n &nbsp; sitt<strong>i</strong>n &nbsp;&nbsp; (substitution of 'e' with 'i')</strong></li>
  <li>sittin &nbsp; sittin<strong>g</strong> &nbsp;&nbsp; (insert 'g' at the end).</strong></li>
</ul>
<i>The Levenshtein distance between &nbsp; "<strong>rosettacode</strong>", &nbsp; "<strong>raisethysword</strong>" &nbsp; is  <strong>8</strong>.</i>
<i>The distance between two strings is same as that when both strings are reversed.</i>
```

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/levenshtein-distance.md
Comment (scissorsneedfoodtoo) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/levenshtein-distance.md]: We can remove the space before the period here:

Write a function that returns the Levenshtein distance between two strings given as parameters.

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/long-multiplication.md
Comment (scissorsneedfoodtoo) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/long-multiplication.md]: This line could be modified slightly and moved to the instructions section below:

```
## Instructions
<section id='instructions'>
Write a function that takes two strings of large numbers as parameters. Your function should return the product of these two large numbers as a string.
<strong>Note:</strong> In JavaScript, arithmetic operations are inaccurate with large numbers, so you will have to implement precise multiplication yourself.
</section>
```

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/last-letter-first-letter.english.md
Comment (scissorsneedfoodtoo) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/last-letter-first-letter.english.md]: function findLongestChain(items) {

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/letter-frequency.english.md
Comment (scissorsneedfoodtoo) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/letter-frequency.english.md]: Same thing here:

`function letterFrequency(txt) {`

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/levenshtein-distance.md
Comment (scissorsneedfoodtoo) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/levenshtein-distance.md]: `function levenshtein(a, b) {`

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/linear-congruential-generator.md
Comment (scissorsneedfoodtoo) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/linear-congruential-generator.md]: `function linearCongGenerator(r0, a, c, m, n) {`

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/long-multiplication.md
Comment (scissorsneedfoodtoo) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/long-multiplication.md]: `function mult(strNum1, strNum2) {`

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/last-letter-first-letter.english.md
Comment (scissorsneedfoodtoo) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/last-letter-first-letter.english.md]: This is not as important as in the seed, but we could probably remove the space here, too:

`function findLongestChain(items) {`

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/letter-frequency.english.md
Comment (scissorsneedfoodtoo) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/letter-frequency.english.md]: `function letterFrequency(txt) {`

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/levenshtein-distance.md
Comment (scissorsneedfoodtoo) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/levenshtein-distance.md]: `function levenshtein(a, b) {`

**Comment by raisedadead** [comment]
Just checking in for a quick confirmation, are these Object Ids hand generated?

**Comment by NitronR** [comment]
@raisedadead Got these from the archived curriculum repo: https://github.com/freeCodeCamp/curriculum/blob/feat/interview-prep/challenges/08-coding-interview-prep/rosetta-code.json

**Comment by raisedadead** [comment]
I see. Thanks for confirming, yeah we would want to have freshly generated ids, but only for any challenge that is being added to the codebase like in this PR. We obviously cannot change the ones already in the repo.

**Comment by NitronR** [comment]
@raisedadead Added new IDs.
@scissorsneedfoodtoo Changes done. Let me know if anything else is required.

**Comment by raisedadead** [comment]
Perfect. Thanks.

**Comment by scissorsneedfoodtoo** [comment]
Thank you for making those changes @NitronR, and for catching the possibly stale ids @raisedadead.

I've asked some other contributors to take a look at this. Once we get one more approval we can merge these in.

**Comment by RandellDawson** [comment]
@scissorsneedfoodtoo Actually, can we get the assert message arguments removed since we have not yet merged this PR.  I don't mind adding another commit and doing it myself.  Also, can we remove the "Good luck" comment as we are removing all unnecessary comments from the challenges for translating to other languages.

**Comment by scissorsneedfoodtoo** [comment]
@RandellDawson, sure, feel free to push your commits to this PR. Thank you for taking care of this so quickly!

**Comment by NitronR** [comment]
@RandellDawson Thanks for the changes. Feel free to review and let me know if anything else is required :)

**Comment by RandellDawson** [comment]
@scissorsneedfoodtoo I just push a commit to remove the assert message arguments for the 5 challenges.  I decided not to go ahead with removing the "Good Luck" comment on this PR.  I will do that on a separate PR since the scope is completely different.

**Review by scissorsneedfoodtoo — APPROVED** [review]

**Comment by scissorsneedfoodtoo** [comment]
@RandellDawson, thank you for making all of those changes and again for your review. Good call on removing the "Good Luck" comment in a separate PR. I'm sure there are others that can be removed, too.

Thanks again for your patience and for these great ports @NitronR! Looking forward to your next PR.

**Comment by RandellDawson** [comment]
@scissorsneedfoodtoo When we get ready to launch this particular section to Chinese, then I will remove the comments in a PR.  We are tackling one certification at a time.

**Comment by scissorsneedfoodtoo** [comment]
@RandellDawson, awesome, that sounds good. Please let me know if there's anything I can do to help with the process.

**Comment by NitronR** [comment]
@scissorsneedfoodtoo @RandellDawson Thanks for your help. I will start working on the next set of problems.

**Pull request merged**
